### DISCUSSION TITLE
Explanation of mosaic and mixup scale?

### DISCUSSION THREAD

**Discussion by andrew-dang** [Megvii-BaseDetection/YOLOX#1072] [Q&A]
Hi there! 

I am trying to get a better understanding of the configuration. I was hoping someone could explain what the values for mosaic_scale (default = (0.1, 2)) and mixup_scale (default = (0.5, 1.5)) are supposed to represent, or what these values do?

Thanks in advance!

**Comment by gouldberg** [comment]
I just add info.

Mosaic is firstly constructed by 4 images (primal 1 image + other randomly selected 3 images,  positioned in 2 * 2).
This is constructed in 1280(H) * 1280(W),  Upper left is primal 1 image and others are randomly selected 3 images.
Base coordinate touching 4 points of 4 images is randomly selected around center of 1280(H) * 1280(W) with some rectangular area.
This mosaic image is translated, sheared ... with some data augment,
Also finally scaled by parameter mosaic_scale,  randomly selected from min = 0.1 to max = 2, if mosaic_scale = (0.1, 2).
After scaling,  upper left 640(H) * 640(W) is finally cropped.
Note that if mosaic parameter 0.1 is applied,  the mosaic image is made to small image, some small objects are not visible depending on the input image.
If mosaic parameter 2 is applied, the mosaic image is made to large,  some large objects are out of upper left 640(H) * 640(W), and only a part of the object is remained after cropped.

Also 1 image is randomly selected to mixup with above mosaic image after cropped.
This selected image is also scaled by mixup_scale parameter.
